envs/environment-tf-py37.yml

```diff
@@ -0,0 +1,143 @@
+name: causalml-tf-py37
+channels:
+  - defaults
+dependencies:
+  - pip=21.0.1
+  - python=3.7
+  - setuptools=52.0.0
+  - graphviz
+  - pip:
+    - absl-py==0.12.0
+    - appnope==0.1.2
+    - argon2-cffi==20.1.0
+    - astunparse==1.6.3
+    - async-generator==1.10
+    - attrs==20.3.0
+    - backcall==0.2.0
+    - bleach==3.3.0
+    - cachetools==4.2.1
+    - cffi==1.14.5
+    - chardet==4.0.0
+    - coverage==5.5
+    - cycler==0.10.0
+    - cython==0.29.23
+    - decorator==5.0.7
+    - defusedxml==0.7.1
+    - dill==0.3.3
+    - entrypoints==0.3
+    - future==0.18.2
+    - gast==0.4.0
+    - google-auth==1.29.0
+    - google-auth-oauthlib==0.4.4
+    - google-pasta==0.2.0
+    - graphviz
+    - grpcio==1.34.0
+    - h5py==3.1.0
+    - idna==2.10
+    - iniconfig==1.1.1
+    - ipykernel==5.5.3
+    - ipython==7.22.0
+    - ipython-genutils==0.2.0
+    - ipywidgets==7.6.3
+    - jedi==0.18.0
+    - jinja2==2.11.3
+    - joblib==1.0.1
+    - jsonschema==3.2.0
+    - jupyter==1.0.0
+    - jupyter-client==6.1.12
+    - jupyter-console==6.4.0
+    - jupyter-contrib-core==0.3.3
+    - jupyter-contrib-nbextensions==0.5.1
+    - jupyter-core==4.7.1
+    - jupyter-highlight-selected-word==0.2.0
+    - jupyter-latex-envs==1.4.6
+    - jupyter-nbextensions-configurator==0.4.1
+    - jupyterlab-pygments==0.1.2
+    - jupyterlab-widgets==1.0.0
+    - keras==2.4.3
```

Comment:
do we need `- keras-preprocessing==1.1.2` here as py36 38 39, too? I searched a bit but actually not sure we are using `keras.preprocessing` though

Reply:
When `tensorflow` is installed, it installs `keras` and `keras-preprocessing` along with `tensorboard` and other `tensorflow` related libraries. We can remove it from the `conda` environment files if we want, because `conda` will install them anyway.